pkg/certificate/conditions.go

```diff
@@ -0,0 +1,106 @@
+package certificate
+
+import (
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MRC condition types
+const (
+	ready              v1alpha2.MeshRootCertificateConditionType = "Ready"
+	accepted           v1alpha2.MeshRootCertificateConditionType = "Accepted"
+	issuingRollout     v1alpha2.MeshRootCertificateConditionType = "IssuingRollout"
+	validatingRollout  v1alpha2.MeshRootCertificateConditionType = "ValidatingRollout"
+	issuingRollback    v1alpha2.MeshRootCertificateConditionType = "IssuingRollback"
+	validatingRollback v1alpha2.MeshRootCertificateConditionType = "ValidatingRollback"
+)
+
+// MRC condition statuses
+const (
+	trueStatus    v1alpha2.MeshRootCertificateConditionStatus = "True"
+	falseStatus   v1alpha2.MeshRootCertificateConditionStatus = "False"
+	unknownStatus v1alpha2.MeshRootCertificateConditionStatus = "Unknown"
+)
+
+// MRC condition reasons
+const (
+	pendingReason = "Pending"
+
+	// Accepted type
+	certificateAcceptedReason = "CertificateAccepted"
+	errorRetrievingCAReason   = "ErrorRetrievingCA"
+
+	// ValidatingRollout type
+	passiveStateValidatingReason      = "PassiveState"
+	passivelyInUseForValidatingReason = "CertificatePassivelyInUseForValidation"
+	notAcceptedValidatingReason       = "NotAccepted"
+
+	// IssuingRollout type
+	passiveStateIssuingReason      = "PassiveState"
+	passivelyInUseForIssuingReason = "CertificatePassivelyInUseForIssuing"
+	notAcceptedIssuingReason       = "NotAccepted"
+
+	// ValidatingRollback type
+	isNotReadyValidatingReason    = "IsNotReady"
+	passiveStateCertTrustedReason = "PassiveStateAndCertIsTrusted"
+	noLongerValidatingReason      = "NoLongerValidating"
+)
+
+func setMRCCondition(mrc *v1alpha2.MeshRootCertificate, conditionType v1alpha2.MeshRootCertificateConditionType, conditionStatus v1alpha2.MeshRootCertificateConditionStatus, conditionReason string, message string) {
+	newCondition := v1alpha2.MeshRootCertificateCondition{
+		Type:    conditionType,
+		Status:  conditionStatus,
+		Reason:  string(conditionReason),
+		Message: message,
+	}
+
+	now := metav1.NewTime(time.Now())
+	newCondition.LastTransitionTime = &now
+
+	for i, condition := range mrc.Status.Conditions {
+		if condition.Type != newCondition.Type {
+			continue
+		}
+
+		// only update transition time on status change
+		if condition.Status == newCondition.Status {
+			newCondition.LastTransitionTime = condition.LastTransitionTime
+		} else {
+			log.Debug().Str("mrc", namespacedMRCName(mrc)).Msgf("updating last transition time for condition type %s", condition.Type)
```

Comment:
I think the wording is correct, but it is confusing. Above in this method we set the `newCondition`'s `LastTransitionTime` to now. If the condition already exists on the MRC and the status of the `newCondition` and existing `condition` are the same, then the `LastTransitionTime` should not be updated so the `newCondition`'s `LastTransitionTime` is set to the current `LastTransitionTime`. Otherwise, the `LastTransitionTime` will be updated (even though this was already done above. I've refactored the code a bit so this will make more sense.

Reply:
oh, yes sorry I read it wrong the first time.  thanks!